src/canister/platform_orchestrator/src/api/canister_management/recharge_subnet_orchestrator.rs

```diff
@@ -0,0 +1,31 @@
+use ic_cdk::{
+    api::management_canister::main::{deposit_cycles, CanisterIdRecord},
+    caller,
+};
+use ic_cdk_macros::update;
+use shared_utils::constant::SUBNET_ORCHESTRATOR_CANISTER_INITIAL_CYCLES;
+
+use crate::CANISTER_DATA;
+
+#[update]
+async fn recharge_subnet_orchestrator() -> Result<(), String> {
+    let subnet_orchestrator_canister_id = caller();
+    let contains_caller_subnet = CANISTER_DATA.with_borrow(|canister_data| {
+        canister_data
+            .all_subnet_orchestrator_canisters_list
+            .contains(&subnet_orchestrator_canister_id)
+    });
+
+    if !contains_caller_subnet {
+        return Err("Subnet Orchestrator not found".into());
+    }
```

Comment:
I believe we can also refactor these repetitive checks and validations in a single place with the **New Type** pattern. Something I learned to appreciate recently for Type oriented programming.

https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Basically create new types like `SubnetOrchestrator` that accepts your parameters and does the validations inside its `new` method or other related methods. That way you're not duplicating this logic on multiple endpoints but using your narrow types as opposed to using the wide types like the `Principal`.

I am guilty of doing this too, before. I remember I have duplicated validation logic at the beginning of multiple endpoints that can also use this refactor

Reply:
Makes sense. Will make the changes accordingly.

Reply:
made the changes in the latest commit.